test/Spec.hs

```diff
@@ -141,3 +145,26 @@ runSpec = testGroup "runGenState_ and runPrimGenIO_"
 -- | Create a StdGen instance from an Int and pass it to the given function.
 seeded :: (StdGen -> a) -> Int -> a
 seeded f = f . mkStdGen
+
+#if __GLASGOW_HASKELL__ >= 802
```

Comment:
It might be a dumb question. I'd be surprised if that generic deriving mechanism doesn't work for older ghc, so why this  CPP is needed here?

Reply:
AFAIR Generics were introduced in 7.2

Reply:
GHC 8.2 improves UX of `{-# LANGUAGE DeriveAnyClass  #-}`, while older GHCs for some reason require `Uniform ()` instance and ask to put `instance Uniform MyAction` on a separate line. I think adding `Uniform ()` is a good idea anyways, so I did this and removed CPP in f009be5.